grama/models/plane_laminate.py

```diff
@@ -226,16 +226,16 @@ def uniaxial_stress_limit(X):
     Param     = Y[:, 0:4]  # [E1, E2, nu12, G12]
     Theta     = Y[:, 4]
     T         = Y[:, 5]
-    Sigma_max = Y[:, 6:11] # [sig_11_t, sig_11_c, sig_22_t, sig_22_c, sig_12_s]
+    Sigma_max = Y[:, 6:11] # [sig_11_t, sig_22_t, sig_11_c, sig_22_c, sig_12_s]
 
     ## Evaluate stress [\sigma_11, \sigma_22, \sigma_12]_i
     Stresses = Nx * uniaxial_stresses(Param, Theta, T)
 
     ## Construct limit state
     g_limit = np.zeros((k, 5))
-    g_limit[:, (0,1)] = +Sigma_max[:, (0,1)] - Stresses[:, (0,1)]
-    g_limit[:, (2,3)] = +Sigma_max[:, (2,3)] + Stresses[:, (0,1)]
-    g_limit[:, 4]     = +Sigma_max[:, 4]     - np.abs(Stresses[:, 2])
+    g_limit[:, (0,1)] = +1 - Stresses[:, (0,1)] / Sigma_max[:, (0,1)]
+    g_limit[:, (2,3)] = +1 - Stresses[:, (0,1)] / Sigma_max[:, (2,3)] # since sig_11_c, sig_22_c > 0
```

Comment:
I believe the original polarity is correct; the max stress criterion for tension and compression (with the compressive strength coded as a positive variable) is:

`-strength_compression < sigma_applied < +strength_tension`

Re-arranging the first inequality yields a limit state for compression

`0 < +strength_compression + sigma_applied = g_compression`

I do like the idea of normalizing the limit states though; that would give the following:

```suggestion
    g_limit[:, (2,3)] = +1 + Stresses[:, (0,1)] / Sigma_max[:, (2,3)]
```

Let me know if the derivation above seems incorrect.

Reply:
Your suggestion is correct. I was viewing the problem from a design perspective of analyzing a laminate subject to tensile and compressive loads. My suggested change is suitable if we consider the compressive limit states for a compressive load. However, we are considering all (tensile and compressive) limit states under a tensile load instead. Naturally a compressive failure will never occur under a tensile load, at least for most simple laminates under simple loading conditions.